lib/core/services/sort-by-category.service.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SortableByCategoryItem, SortByCategoryMapperService } from './sort-by-category.service';
+
+interface TestSortableByCategoryItem extends SortableByCategoryItem {
+    id: string;
+}
+
+describe('SortByCategoryMapperService', () => {
+
+    let mapper: SortByCategoryMapperService<TestSortableByCategoryItem>;
+
+    const DEFAULT_CATEGORIES = [
+        '',
+        'DefaultCategory1',
+        'DefaultCategory2',
+        'DefaultCategory3'
+    ];
+
+    beforeEach(() => {
+        mapper = new SortByCategoryMapperService();
+    });
+
+    it('should map items by categories ', () => {
+
+        const items: TestSortableByCategoryItem[] = [
+            { id: 'id1', name: 'firstCategory_222', category: 'category1' },
+            { id: 'id2', name: 'secondCategory_AA', category: 'category2' },
+            { id: 'id3', name: 'firstCategory_111', category: 'category1' },
+            { id: 'id4', name: 'secondCategory_BB', category: 'category2' },
+            { id: 'id5', name: 'Default_a', category: DEFAULT_CATEGORIES[1] },
+            { id: 'id6', name: 'Default_b', category: DEFAULT_CATEGORIES[0] }
+        ];
+
+        const expectedItemsByCategory = [
+            { category: 'category1', items: [items[2], items[0]] },
+            { category: 'category2', items: [items[1], items[3]] },
+            { category: '', items: [items[4], items[5]] }
+        ];
+
+        const result = mapper.mapItems(items, DEFAULT_CATEGORIES);
+
+        expect(result).toEqual(expectedItemsByCategory);
+    });
+
+    it('should set all items under default category', () => {
+
+        const defaulValues: TestSortableByCategoryItem[] = [{
+            name: 'name-a',
+            id: 'id',
+            category: DEFAULT_CATEGORIES[1]
+        }, {
+            name: 'name-b',
+            id: 'id2',
+            category: DEFAULT_CATEGORIES[2]
+        }, {
+            name: 'name-c',
+            id: 'id3',
+            category: DEFAULT_CATEGORIES[0]
+        }];
+
+        const result = mapper.mapItems(defaulValues, DEFAULT_CATEGORIES);
+
+        expect(result.length).toBe(1);
+        expect(result[0].category).toBe('');
+        expect(result[0].items.length).toBe(defaulValues.length);
+    });
+
+    it('should set all items under specific category if at least one item has category', () => {
```

Comment:
probably is better to provide an example or have the expected array as result. It not clear what we are testing

Reply:
After checking this test,  I see that the expected array as a result is already verified in the test case --> 
 `should map items by categories` .
Then I think it is better to remove it.